stdlib/args.ysh

```diff
@@ -60,10 +60,23 @@ proc arg-parse (; place ; ; block_def) {
     setvar p.rest = null
   }
 
+  var names = {}
+  for items in ([p.flags, p.args]) {
+    for x in (items) {
+      if (x.name in names) {
+        error "Duplicate flag/arg name $[x.name] in spec" (status=3)
```

Comment:
Aside - I wonder if `(status=3)` is hard to remember (for us, not for users, where it doesn't really matter)

I think in the Python code it's easy enough to to do `raise error.Expr` or `raise error.TypeErr`

I think the main reason I created the status 3 is because bash/dash doesn't use that code, so it makes spec  tests a little more distinctive